c_emulator/riscv_sim.c

```diff
@@ -217,7 +217,7 @@ static void read_dtb(const char *path)
   fprintf(stdout, "Read %zd bytes of DTB from %s.\n", dtb_len, path);
 }
 
-char *process_args(int argc, char **argv)
+int process_args(int argc, char **argv)
```

Comment:
This change refactors the process_args signature (although motivated by the new use-case).
Please split off into a separate, standalone commit (together with the change below that changes the usage of process_args).

Could you also add a comment above the process_args function to document its contract?